docs/source/architecture/events_and_transactions_receipts.rst

```diff
@@ -0,0 +1,343 @@
+  re-executing it.
+
+Sawtooth transaction receipts are represented as a protobuf message when exposed
+to clients. They contain a list of StateChange messages, a list of Event
+messages, and a list of TransactionReceipt.Data messages:
+
+.. code-block:: protobuf
+
+  message TransactionReceipt {
+    message Data {
+      // Interpretation left to transaction family
+      string data_type = 1;
+      bytes data = 2;
+    }
+    // State changes made by this transaction
+    // StateChange is already defined in protos/state_delta.proto
+    repeated StateChange state_changes = 1;
+    // Events fired by this transaction
+    repeated Event events = 2;
+    // Transaction family defined data
+    repeated Data data = 3;
+
+    string transaction_id = 4;
+  }
+
+The fields in the TransactionReceipt.Data are opaque to Sawtooth and their
+interpretation is left up to the transaction family. TransactionReceipt.Data
+can be requested for a transaction that has been committed and should only be
+used to store information about a transaction’s execution that should not be
+kept in state. Events produced by transaction processors can be subscribed to
+by clients using the event system described above.
+
+Transaction Receipt Store
+-------------------------
+
+The TransactionReceiptStore class stores receipts. The TransactionReceiptStore
+will be written to by the ChainController when a block is validated and the
+ChainController will be responsible for constructing the TransactionReceipt
+protobuf message.
+
+Transaction receipts will only be stored in this off-chain store and will not
+be included in the block. Note that because a transaction may exist in multiple
+blocks at a time, the transaction receipt is stored by both transaction id and
+block state root hash.
+
+.. code-block:: python
+
+  class TransactionReceiptStore:
+  	def put_receipt(self, txn_id, receipt):
+      	"""Add the given transaction receipt to the store. Does not guarantee
+         	it has been written to the backing store.
+
+      	Args:
+          	txn_id (str): the id of the transaction being stored.
+          	state_root_hash: the state root of the block this transaction was
+            executed in.
+          	receipt (TransactionReceipt): the receipt object to store.
+      	"""
+
+  	def get_receipt(self, txn_id):
+      	"""Returns the TransactionReceipt
+
+      	Args:
+          	txn_id (str): the id of the transaction for which the receipt
+            should be retrieved.
+          	state_root_hash: the state root of the block this transaction was
+            executed in.
+
+      	Returns:
+          	TransactionReceipt: The receipt for the given transaction id.
+
+      	Raises:
+          	KeyError: if the transaction id is unknown.
+      	"""
+
+Message Handlers
+----------------
+
+Once transaction receipts are stored in the TransactionReceiptStore, clients
+need to be able to request a transaction receipt for a given transaction id. To
```

Comment:
Can you change "need to be able to" to "can"?  That is, does "clients can request a transaction receipt..." mean the same thing?  If so, my rule # 2 is that fewer words are better.